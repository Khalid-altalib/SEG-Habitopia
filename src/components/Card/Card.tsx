// React Native
import { View } from "react-native";

// Native Base
import { theme, useColorModeValue } from "native-base";
import { StyleSheet } from "react-native";

// Expo Linear Gradient
import { LinearGradient } from "expo-linear-gradient";

type Props = {
  /** The background color gradient of the card. If undefined, then default colors are used. */
  backgroundColorGradient?: [string, string];
  /** The children to be displayed within the card. */
  children: any;
};

/**
 * A card which can contain children.
 *
 * @param props The properties passed to the component.
 * @returns The component representing the card and its children.
 */
const Card = ({ backgroundColorGradient, children }: Props) => {
  const LIGHT_MODE_GRADIENT = [theme.colors.white, theme.colors.gray[100]];
  const DARK_MODE_GRADIENT = [
    theme.colors.blueGray[700],
    theme.colors.blueGray[600],
  ];

  const styles = StyleSheet.create({
    base: {
      borderRadius: 10,
      padding: 20,
    },
  });

  return (
    <View
      style={{
        shadowColor: "black",
        shadowOffset: { width: 5, height: 5 },
        shadowOpacity: useColorModeValue(0.05, 0.15),
      }}
    >
      <LinearGradient
        colors={
          backgroundColorGradient ||
          useColorModeValue(LIGHT_MODE_GRADIENT, DARK_MODE_GRADIENT)
        }
        style={[styles.base]}
        end={{ x: 0.5, y: 2 }}
      >
        {children}
      </LinearGradient>
    </View>
  );
};

export default Card;
