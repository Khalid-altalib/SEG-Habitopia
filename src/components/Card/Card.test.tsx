// React
import React from "react";

// React Native Testing Library
import { render, screen } from "@testing-library/react-native";

// Habitopia
import Card from "@components/Card";

it("Renders correctly.", () => {
  render(<Card>This is some example content.</Card>);

  expect(screen.toJSON()).toMatchSnapshot();
});
