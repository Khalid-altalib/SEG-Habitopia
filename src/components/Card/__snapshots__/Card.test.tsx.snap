// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly. 1`] = `
<View
  style={
    {
      "shadowColor": "black",
      "shadowOffset": {
        "height": 5,
        "width": 5,
      },
      "shadowOpacity": 0.05,
      "width": "100%",
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      {
        "colors": [
          4294967295,
          4294243573,
        ],
        "endPoint": [
          0.5,
          2,
        ],
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      [
        {
          "borderRadius": 10,
          "padding": 20,
        },
      ]
    }
  >
    This is some example content.
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
